### PR TITLE
Reformatted main samples README for better clarity

### LINKED ISSUE #2541: Main samples readme review
We need to review the main samples index.  Specifically.

* Ensure all sample are present and links are correct. e.g. Sample 81 is missing.
* I think it is worth considering changing the 'View' links to either an icon or language text. e.g. 'C#' or 'JavaScript'. This is because you cannot see the column headers
* Consider if it is feasible / beneficial to separate into multiple lists, with skip links at the top. e.g. General / Cognitive Services / Teams / Adapters etc.

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #2541

- Split list into groups
- Provided links to each group
- Reduced columns to: .NET, JavaScript, Python

Possible TODO: Provide friendly names instead of directory names.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,60 +1,101 @@
...
-|56.teams-file-upload     | Demonstrates how to obtain file consent, and upload files to Teams from a bot. Also, how to receive a file sent to a bot.|[View][cs#56]|[View][js#56]          |              |     |     |[View][py#56]
-|57.teams-conversation-bot     | Demonstrates various features of bots on Teams: message all members in a Team or Channel, @mention a user from a bot, update previously sent messages, etc. |[View][cs#57]|[View][js#57]          |              |     |     |[View][py#57]
-|70.styling-webchat     | This sample shows how to create a web page with custom Web Chat component.|         |          |              |  [View][es#70] |     |
-|80.skills-simple-bot-to-bot     | This sample shows how to connect a skill to a skill consumer.| [View][cs#80] | [View][js#80]       |              |   |     |[View][py#80]
+## Sample lists
+
+Samples are designed to illustrate scenarios you'll need to implement to build great bots!
+
+- [Bot Essentials](#bot-essentials)
+- [Bot Advanced](#bot-advanced)
```
Comment (JonathanFingold) [inline README.md:24]: Perhaps move 16.proactive-messages, 17.multilingual-bot, and 44.prompt-user-for-input to the **Advanced bots** section

File: README.md [hunk 2]
```diff
@@ -1,60 +1,101 @@
...
+- [Auth Samples](#auth-samples)
+- [QnAMaker Samples](#qnamaker-samples)
+- [Teams Samples](#teams-samples)
+- [Skills Samples](#skills-samples)
+
+### Bot Essentials
+
+| Sample Name           | Description                                                                      | .NET    | JavaScript   | Python  |
+|:----------------------|:---------------------------------------------------------------------------------|:--------|:-------------|:--------|
+|2.echo-bot             | Demonstrates how to receive and send messages.                                 |[Core][cs#2] |[JS][js#2], [TS][ts#2]|[Python][py#2]
```
Comment (JonathanFingold) [inline README.md:34]: I thought the suggestion had been to create one column for sample number and one for a friendly short name:
```suggestion
| Number | Name | Description | .NET | JavaScript | Python  |
| :----- | :--- | :----------- | :--- | :-------- | :----- |
| 2 | Echo bot | Demonstrates how to receive and send messages. | [.NET Core][cs#2] | [JavaScript][js#2], [TypeScript][ts#2] | [Python][py#2]

```
And so on

File: README.md [hunk 3]
```diff
@@ -1,60 +1,101 @@
...
+
+### Auth Samples
+
+| Sample Name           | Description                                                                      | .NET    | JavaScript   | Python  |
+|:----------------------|:---------------------------------------------------------------------------------|:--------|:-------------|:--------|
+|18.bot-authentication  | Bot that demonstrates how to integrate OAuth providers.                  |[Core][cs#18]|[JS][js#18]|[Python][py#18]
+|24.bot-auth-msgraph    | Demonstrates bot authentication capabilities of Azure Bot Service. Demonstrates utilizing the Microsoft Graph API to retrieve data about the user.|[Core][cs#24] |[JS][js#24] |[Python][py#24]
+|46.teams-auth    | Demonstrates how to use authentication for a bot running in Microsoft Teams. | [Core][cs#46] | [JS][js#46]   |[Python][py#46]
+
+### QnAMaker Samples
```
Comment (JonathanFingold) [inline README.md:71]:
```suggestion
### QnA Maker samples

```

File: README.md [hunk 4]
```diff
@@ -1,60 +1,101 @@
...
+
+| Sample Name           | Description                                                                      | .NET    | JavaScript   | Python  |
+|:----------------------|:---------------------------------------------------------------------------------|:--------|:-------------|:--------|
+|11.qnamaker              | Demonstrates how to use QnA Maker to have simple single-turn conversations     |[Core][cs#11]|[JS][js#11]      |[Python][py#11]
+|49.qnamaker-all-features | Demonstrates how to integrate Multiturn and Active learning in a QnA Maker bot.  This also demonstrates the QnAMakerDialog class. |[Core][cs#49]|[JS][js#49]      |
+
+### Teams Samples
```
Comment (JonathanFingold) [inline README.md:78]:
```suggestion
### Teams samples

```

File: README.md [hunk 5]
```diff
@@ -1,60 +1,101 @@
...
+|53.teams-messaging-extensions-action-preview     | Demonstrates how to create a Preview and Edit flow for a Messaging Extension.|[Core][cs#53]|[JS][js#53]          |[Python][py#53] |
+|54.teams-task-module     | Demonstrates how to retrieve a Task Module, and values from cards in the Task Module, for a Messaging Extension.|[Core][cs#54]|[JS][js#54]          |[Python][py#54] |
+|55.teams-link-unfurling     | A Messaging Extension that performs link unfurling.|[Core][cs#55]|[JS][js#55]          |[Python][py#55] |
+|56.teams-file-upload     | Demonstrates how to obtain file consent, and upload files to Teams from a bot. Also, how to receive a file sent to a bot.|[Core][cs#56]|[JS][js#56]          |[Python][py#56] |
+|57.teams-conversation-bot     | Demonstrates various features of bots on Teams: message all members in a Team or Channel, @mention a user from a bot, update previously sent messages, etc. |[Core][cs#57]|[JS][js#57]          |[Python][py#57] |
+|58.teams-start-new-thread-in-channel     | Demonstrates creating a new thread in a channel. |[Core][cs#5]|[JS][js#58]          |[Python][py#58] |
+
+### Skills Samples
```
Comment (JonathanFingold) [inline README.md:93]:
```suggestion
### Skills samples

```

File: README.md [hunk 6]
```diff
@@ -1,60 +1,101 @@
...
-|55.teams-link-unfurling     | A Messaging Extension that performs link unfurling.|[View][cs#55]|[View][js#55]          |              |     |     |[View][py#55]
-|56.teams-file-upload     | Demonstrates how to obtain file consent, and upload files to Teams from a bot. Also, how to receive a file sent to a bot.|[View][cs#56]|[View][js#56]          |              |     |     |[View][py#56]
-|57.teams-conversation-bot     | Demonstrates various features of bots on Teams: message all members in a Team or Channel, @mention a user from a bot, update previously sent messages, etc. |[View][cs#57]|[View][js#57]          |              |     |     |[View][py#57]
-|70.styling-webchat     | This sample shows how to create a web page with custom Web Chat component.|         |          |              |  [View][es#70] |     |
-|80.skills-simple-bot-to-bot     | This sample shows how to connect a skill to a skill consumer.| [View][cs#80] | [View][js#80]       |              |   |     |[View][py#80]
+## Sample lists
+
+Samples are designed to illustrate scenarios you'll need to implement to build great bots!
```
[the remark this answers is not in the document]
Comment (tracyboehrer) [reply README.md:21]: Aspects?

[the remark this answers is not in the document]
Comment (tracyboehrer) [reply README.md:21]: Functionality?

Samples are designed to illustrate aspects you'll need to implement to build great bots!

Samples are designed to illustrate functionality you'll need to implement to build great bots!

File: README.md (hunk 1 above)
Comment (tracyboehrer) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (tracyboehrer) [reply README.md:36]: And that makes sense... Except for the convenient duplication of 

1.console-echo and 1.browser-echo

[the remark this answers is not in the document]
Comment (tracyboehrer) [reply README.md:36]: I'll commit these link name changes so you can compare the difference.  Easy to change back.

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JonathanFingold) [reply README.md]: I think either of those would work.

File: README.md (hunk 2 above)
Comment (JonathanFingold) [reply README.md]: Ack. :(

File: README.md [hunk 7]
```diff
@@ -1,60 +1,104 @@
...
+|21.corebot-app-insights     | Demonstrates how to add telemetry logging to your bot, storing telemetry within Application Insights.|[.NET Core][cs#21] |[JavaScript][js#21] |
+|23.facebook-events     | Integrate and consume Facebook specific payloads, such as post-backs, quick replies and opt-in events.|[.NET Core][cs#23] |[JavaScript][js#23] |[Python][py#23]
+|42.scaleout            | Demonstrates how you can build your own state solution from the ground up that supports scaled out deployment with ETag based optimistic locking. |[.NET Core][cs#42] |    |[Python][py#42]
+|44.prompt-for-user-input | Demonstrates how to implement your own _basic_ prompts to ask the user for information. |[.NET Core][cs#44]|[JavaScript][js#44]|[Python][py#44]
+|47.inspection    | Demonstrates how to use middleware to allow the Bot Framework Emulator to debug traffic into and out of the bot in addition to looking at the current state of the bot. | [.NET Core][cs#47] | [JavaScript][js#47]   |[Python][py#47]
+|70.styling-webchat     | This sample shows how to create a web page with custom Web Chat component.|         | [ECMAScript 6][es#70]         |
+
+### Authentication Samples
```
Comment (JonathanFingold) [inline README.md:63]:
```suggestion
### Authentication samples

```

**Comment by JonathanFingold** [comment]
I wonder if you can (and it would make sense to) put in non-breaking spaces in the .NET Core and .NET Web text, such as
`[.NET&nbsp;Core][...]`.

**Comment by tracyboehrer** [comment]
> I wonder if you can (and it would make sense to) put in non-breaking spaces in the .NET Core and .NET Web text, such as
> `[.NET&nbsp;Core][...]`.

That works

**Comment by garypretty** [comment]
@tracyboehrer This does look awesome. Going to merge this now - I will raise an additional PR today to add the adapter samples into the list as an additional section and to probably amend the sample names to be friendly.

**Pull request merged**